integrationTests/chainSimulator/mempool/mempool_test.go

```diff
@@ -485,3 +486,874 @@ func TestMempoolWithChainSimulator_Eviction(t *testing.T) {
+		NumItemsToPreemptivelyEvict: 1,
+		TxCacheBoundsConfig: config.TxCacheBoundsConfig{
+			MaxNumBytesPerSenderUpperBound: maxNumBytesPerSenderUpperBoundTest,
+		},
+	}, host)
+
+	require.Nil(t, err)
+	require.NotNil(t, txpool)
+
+	// assure we have enough balance for each account
+	selectionSession := createMockSelectionSession(map[string]*accountInfo{
+		"alice": {
+			balance: oneEGLD,
+			nonce:   0,
+		},
+		"bob": {
+			balance: oneEGLD,
+			nonce:   0,
+		},
+		"receiver": {
+			balance: big.NewInt(0),
+			nonce:   0,
+		},
+	})
+
+	options := holders.NewTxSelectionOptions(
+		10_000_000_000,
+		2,
+		selectionLoopMaximumDuration,
+		10,
+	)
+
+	nonceTracker := newNoncesTracker()
+	transactions := make([]*transaction.Transaction, 0)
+
+	transactions = append(transactions, &transaction.Transaction{
+		Nonce:     nonceTracker.getThenIncrementNonceByStringAddress("alice"),
+		Value:     oneQuarterOfEGLD,
+		SndAddr:   []byte("alice"),
+		RcvAddr:   []byte("receiver"),
+		Data:      []byte{},
+		GasLimit:  50_000,
+		GasPrice:  1_000_000_000,
+		ChainID:   []byte(configs.ChainID),
+		Version:   2,
+		Signature: []byte("signature"),
+	})
+
+	transactions = append(transactions, &transaction.Transaction{
+		Nonce:     nonceTracker.getThenIncrementNonceByStringAddress("bob"),
+		Value:     oneQuarterOfEGLD,
+		SndAddr:   []byte("bob"),
+		RcvAddr:   []byte("receiver"),
+		Data:      []byte{},
+		GasLimit:  50_000,
+		GasPrice:  1_000_000_000,
+		ChainID:   []byte(configs.ChainID),
+		Version:   2,
+		Signature: []byte("signature"),
+	})
+
+	transactions = append(transactions, &transaction.Transaction{
+		Nonce:     nonceTracker.getThenIncrementNonceByStringAddress("alice"),
+		Value:     oneQuarterOfEGLD,
+		SndAddr:   []byte("alice"),
+		RcvAddr:   []byte("receiver"),
+		Data:      []byte{},
+		GasLimit:  50_000,
+		GasPrice:  1_000_000_000,
+		ChainID:   []byte(configs.ChainID),
+		Version:   2,
+		Signature: []byte("signature"),
+	})
+
+	transactions = append(transactions, &transaction.Transaction{
+		Nonce:     nonceTracker.getThenIncrementNonceByStringAddress("bob"),
+		Value:     oneQuarterOfEGLD,
+		SndAddr:   []byte("bob"),
+		RcvAddr:   []byte("receiver"),
+		Data:      []byte{},
+		GasLimit:  50_000,
+		GasPrice:  1_000_000_000,
+		ChainID:   []byte(configs.ChainID),
+		Version:   2,
+		Signature: []byte("signature"),
+	})
+
+	txHashes := make([][]byte, 0)
+	for i, tx := range transactions {
+		txHash := []byte(fmt.Sprintf("txHash%d", i))
+		txHashes = append(txHashes, txHash)
+		txpool.AddTx(&txcache.WrappedTransaction{
+			Tx:               tx,
+			TxHash:           txHash,
+			SenderShardID:    0,
+			ReceiverShardID:  0,
+			Size:             0,
+			Fee:              big.NewInt(int64(tx.GasLimit * tx.GasPrice)),
+			PricePerUnit:     0,
+			TransferredValue: tx.Value,
+			FeePayer:         tx.SndAddr,
+		})
+	}
+
+	blockBody := block.Body{MiniBlocks: []*block.MiniBlock{
+		{
+			TxHashes: txHashes[:len(txHashes)/2],
```

Comment:
Maybe do a small test utility function to create a block body object from a list (or range of list) of transactions? Can be used in other similar tests, as well.